pairing/bn256/suite_test.go

```diff
@@ -27,25 +28,31 @@ func TestScalarMarshal(t *testing.T) {
 func TestScalarOps(t *testing.T) {
 	suite := NewSuite()
 	a := suite.G1().Scalar().Pick(random.New())
-	b := suite.G1().Scalar()
-	b.Add(a, b)
-	a.Sub(a, b)
-	if !a.Equal(suite.G1().Scalar().Zero()) {
-		t.Fatal("bn256: add sub failed")
-	}
-	a.Pick(random.New())
-	b.Set(suite.G1().Scalar().One())
-	b.Mul(a, b)
-	a.Div(a, b)
-	if !a.Equal(suite.G1().Scalar().One()) {
-		t.Fatal("bn256: mul div failed")
-	}
-	a.Pick(random.New())
-	b.Inv(a)
-	a.Mul(a, b)
-	if !a.Equal(suite.G1().Scalar().One()) {
-		t.Fatal("bn256: inversion failed")
-	}
+	b := suite.G1().Scalar().Pick(random.New())
+	c := suite.G1().Scalar().Pick(random.New())
+	d := suite.G1().Scalar()
+	e := suite.G1().Scalar()
+	d.Add(a, b)
+	d.Sub(d, c)
+	e.Sub(a, c)
+	e.Add(e, b)
+	require.True(t, d.Equal(e))
+	d.One()
+	e.One()
+	d.Mul(a, b)
+	d.Div(d, c)
+	e.Div(a, c)
+	e.Mul(e, b)
+	require.True(t, d.Equal(e))
+	d.One()
+	e.One()
+	d.Mul(a, b)
+	d.Mul(d, c)
+	d.Inv(d)
+	e.Mul(a, b)
+	e.Mul(e, c)
+	e.Mul(e, d)
+	require.True(t, e.Equal(suite.G1().Scalar().One()))
```

Comment:
For these kind of code I'd like to see an equation in the comments to explain what is happening.

Reply:
Done.